src/include/kdbtypes.h

```diff
@@ -48,8 +48,44 @@ using octet_t = uint8_t; // default: 0
 
 // for C (and C++)
 
-typedef unsigned char kdb_boolean_t;
 typedef unsigned char kdb_char_t;
+
+#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
+// for C99+
+#include <stdbool.h>
+#include <stdint.h>
+
+typedef uint8_t kdb_octet_t;
+typedef bool kdb_boolean_t;
+typedef int16_t kdb_short_t;
+typedef int32_t kdb_long_t;
+typedef int64_t kdb_long_long_t;
+typedef uint16_t kdb_unsigned_short_t;
+typedef uint32_t kdb_unsigned_long_t;
+typedef uint64_t kdb_unsigned_long_long_t;
+
+#if SIZEOF_LONG == 4
```

Comment:
The `#if` is not necessary. In C99 inttypes.h defines macros for the printf modifiers, e.g., PRIu8

Reply:
Great, I didn't know about these!